_rules/inline_link_in_paragraph_is_distinguishable_be4d0c.md

```diff
@@ -0,0 +1,240 @@
+---
+id: be4d0c
+name: Inline link in paragraph is distinguishable
+rule_type: atomic
+
+description: |
+  This rule checks that links that are embedded in a paragraph have a way to distinguish them as links compared to the rest of the text
+
+accessibility_requirements: # Remove whatever is not applicable
+  wcag20:1.4.1: # Use of Color (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+
+acknowledgements:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+---
+
+## Applicability
+
+This rule applies to any HTML element which:
+
+- has the [semantic role][] of ['link'][link] or a [semantic role][] that inherits from the ['link'][link] role; and
+- is a [descendant][] of a `p` element; and
+- has [descendant][] [text nodes][text node] that are [visible][]; and
+- the `p` element has other [descendant][] [visible][] [text nodes][text node].
+
+## Expectation
+
+Each target element has either:
+
+- An icon immediately before, after or in the link denoting it as a link
+- Text immediately before, after or in the link denoting it as a link
+- a [distinguishing style][] compared to the other text not based on color alone
+- a different color compared to the rest of the text in the paragraph that has at least a 3:1 [contrast ratio](https://www.w3.org/TR/WCAG21/#dfn-contrast-ratio) difference with the rest of the text and a [distinguishing style][] both when the target element [gains focus][gain focus] and the target element is [hovered][]
+
+## Assumptions
+
+This rule assumes that the link is distinguishable from the rest of the text with color, which means it fails SC 1.4.1 when there is not another way to distinguish it.
```

Comment:
I don't understand this assumption. The expectation tests for color.

Reply:
The expectation is indeed testing for color, but it needs to be a contrast ratio of at least 3:1 for the expectation to be met.

If we leave this assumption out of it; there could be a scenario where the link is not(!) a different color from the rest of the text and this rule would fail it even though it should not fail the SC. (Because you should only fail this SC if you communicate something with color without communicating it in a different way.

Does that help you understand this assumption?

Reply:
indeed, if the link is the same color as the surrounding text, this rule is not applicable (it's an odd gap in WCAG since in those cases it's deemed that the link is indistinguishable for all sighted users, not just those with color deficiency/vision impairment that requires at least a 3:1 contrast - though paradoxically keyboard/AT users are advantages because THEY can get to the links without any problem)